### PR TITLE
Move token extract logic to SDK

### PR DESCRIPTION
**Pull request by gouthamMN**
**What does this PR do?**
-  Add helper function to extract `oid`, `_claim_names`, & `groups` from a given JWT token.
- Update func `CreateAuthorizationRequest()` to pass `token` instead of `SubjectAttributes`.
- Refactor unit tests to remove some ARO-RP test util imports.
- Remove `generate.go` file as we don't generate and mocks.

**Testing**
- New Unit tests are added.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
       - name: Set up Golang
         uses: actions/setup-go@0a12ed9d6a96ab950c8f026ed9f722fe0da7ef32 # v5.0.2
         with:
-          go-version: "1.22.0"
+          go-version: "1.21.0"
```
Comment (stevekuznetsov) [inline .github/workflows/ci.yml:27]: Why?

File: client/client_test.go [hunk 2]
```diff
@@ -10,17 +10,17 @@ import (
...
-	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
-	"github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
 	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/checkaccess-v2-go-sdk/client/internal/test"
 	"github.com/google/go-cmp/cmp"
+)
 
-	testhttp "github.com/Azure/ARO-RP/test/util/http/server"
+var (
+	endpoint = "https://westus.authorization.azure.net/providers/Microsoft.Authorization/checkAccess?api-version=2021-06-01-preview"
```
Comment (stevekuznetsov) [inline client/client_test.go:19]: Global variables + parallel test runs = bad time. Why should these be anywhere but in the test that uses them? There's no real thing as DRY for test data, if two tests need this and declare it, it's fine.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (gouthamMN) [reply .github/workflows/ci.yml:27]: I downgraded SDK go to support `1.21` as ARO-RP still on `1.21`. Updated the go version to keep it consistent.

File: client/internal/token/token_test.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+			got, err := ExtractClaims(tt.token)
+			t.Log(got, err)
+
+			if tt.wantErr {
+				if err == nil {
+					t.Errorf("Expect an error but got nothing")
+				}
+			} else {
+				if got.ObjectId != tt.wantOid {
+					t.Errorf("Got oid: %q, want %q", got.ObjectId, tt.wantOid)
```
Comment (stevekuznetsov) [inline client/internal/token/token_test.go:54]: Why not just `cmp.Diff()` the whole `got`?

**Review by stevekuznetsov — COMMENTED** [review]
Looks good, couple comments.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (stevekuznetsov) [reply .github/workflows/ci.yml:27]: Gotcha

File: client/client_test.go (hunk 2 above)
Comment (gouthamMN) [reply client/client_test.go:19]: Just thought of reducing some lines in each test, nothing more :) I will make the suggested changes.

File: client/internal/token/token_test.go (hunk 3 above)
Comment (gouthamMN) [reply client/internal/token/token_test.go]: updated

File: client/client_test.go (hunk 2 above)
Comment (gouthamMN) [reply client/client_test.go]: updated

**Review by stevekuznetsov — APPROVED** [review]

**Pull request merged**